python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -85,75 +83,151 @@ def __init__(
         ):
             self._reader_registered = True
 
+        self._meta_channel: Optional[Channel] = _meta_channel
+        if (
+            self._meta_channel is None
+            and self._writer_registered
+            and (
+                self._typ.shape == TorchTensorType.AUTO
+                or self._typ.dtype == TorchTensorType.AUTO
+            )
+        ):
+            # Allocate 1KiB for metadata.
+            metadata_type = SharedMemoryType(buffer_size_bytes=1_000)
+            self._meta_channel = metadata_type.create_channel(
+                self._writer,
+                self._readers,
+            )
+
+        if self._meta_channel is None:
+            # If there is no metadata channel, then we can only pass tensors of
+            # static shape and dtype.
+            assert self.has_static_type()
```

Comment:
Can this happen by a user error? Maybe raise error instead?

Reply:
Ah no this is developer bug.